### ISSUE TITLE
Memory consumption estimation

### ISSUE THREAD

**Issue by karel-brinda** [jermp/fulgor#43]
Hello,

What's the best way to estimate Fulgor memory consumption for a given index? Is it the disk size plus some additional constant space (buffers, etc.)?

This info would be extremely helpful for job planning. 

Thanks!!

**Comment by jermp** [comment]
Hi Karel,
what do you mean exactly:
1. given a raw collection of files, try to guess the final Fulgor index size? or,
2. given an already built Fulgor index on disk, compute its memory at query time?

Regarding 1., this is very hard to tell because it depends a lot on the characteristics of the collection itself.

Regarding 2., this is very easy instead :) as the size of Fulgor on disk is essentially that used during queries (+ some very small buffer per thread).

See the following two tables. The first reports space on disk, the second RAM usage at query time.

<img width="691" alt="Screen Shot 2024-11-12 at 08 59 25" src="https://github.com/user-attachments/assets/a9634fc6-1cec-4341-907d-71eb8f3fc5b1">

<img width="625" alt="Screen Shot 2024-11-12 at 08 59 10" src="https://github.com/user-attachments/assets/cc69a592-aab4-40aa-b5ff-bc70b7fb8c12">

**Comment by karel-brinda** [comment]
Yes, I meant 2. Thanks for the table + explanation. So is disk_size +1GB always safe as the upper limit?

**Comment by jermp** [comment]
Yup, I'd say so: disk size + 1 extra GB looks a reasonable upper bound :)

Yes, a formula is possible, but it does also depend on the number of distinct color set ids we see
in a query, which obviously varies with the query. CC @Alessio-Campa I'm also pinging Alessio here.
